internal/runbits/buildscript/buildscript.go

```diff
@@ -83,21 +83,22 @@ func Sync(proj *project.Project, commitID *strfmt.UUID, out output.Outputer, aut
 			Script:       expr,
 		})
 		if err != nil {
-			return errs.Wrap(err, "Could not update project to reflect build script changes.")
+			return false, errs.Wrap(err, "Could not update project to reflect build script changes.")
 		}
+		staged = true
 		commitID = &stagedCommitID
 
 		expr, err = getBuildExpression(proj, commitID, auth) // timestamps might be different
 		if err != nil {
-			return errs.Wrap(err, "Could not get remote build expr for staged commit")
+			return false, errs.Wrap(err, "Could not get remote build expr for staged commit")
```

Comment:
Technically it was staged at this point, but there was an error fetching the resulting expression. Would it be bad to change this `false` to `staged`?

Reply:
Good question, I'm not exactly sure. If the user ends up in a state where that staging is irrelevant/useless, then I'd argue that false is appropriate. If there is some utility to the staged commit, then I'd argue true. If we land on true while an error has occurred, then I think the caller will need to communicate more clearly to the user what is going on.

Reply:
To be safe, I would return `staged` here (and perhaps everywhere for clarity). It will be up to the caller to handle the distinction.

Reply:
I think false is appropriate. It does feel pointless to succeed on something we know won't work. If there is indeed a use-case for this let's hear about it before we address it.

Reply:
That said; maybe rename `staged` to `success` or something? It does seem that the name is a little off given it's not actually communicating whether it was staged (not JUST whether it was staged, anyway).